### PR TITLE
balena deploy: Fix "TypeError: images.push is not iterable"

### PR DESCRIPTION
**Pull request by pdcastro**
Resolves: #1783 
Change-type: patch

Internal discussion: https://www.flowdock.com/app/rulemotion/i-cli/threads/GOcVcJdg2SN3R_6VIIlt1LnPObL

### CODE REVIEW

File: lib/utils/compose.js [hunk 1]
```diff
@@ -519,7 +519,7 @@ const getPreviousRepos = (sdk, docker, logger, appID) =>
 		})
 		.then(function(release) {
 			// grab all images from the latest release, return all image locations in the registry
-			if (release.length > 0) {
+			if (release && release.length > 0) {
```
Comment (Page-) [inline lib/utils/compose.js:522]: The release && should be unnecessary as the line client should never return null for this

Comment (pdcastro) [reply lib/utils/compose.js:522]: > **should be** unnecessary [...] pine client **should** never

_vs._

> **is** unnecessary [...] pine client **will** never

That's OK, I am happy with the harmless redundancy when it gives me extra confidence. :-)

Comment (Page-) [reply lib/utils/compose.js:522]: If it ever gives you anything other than an array things have gone so badly wrong that bailing is actually the best you can hope for

File: lib/utils/compose.js [hunk 2]
```diff
@@ -530,6 +530,8 @@ const getPreviousRepos = (sdk, docker, logger, appID) =>
 						return registry.imageName;
 					});
 				});
+			} else {
+				return Promise.resolve([]);
 			}
 		})
 		.catch(e => {
```
Comment (srlowe) [inline lib/utils/compose.js:537]: Wont we get an undefined returned here too?

Comment (Page-) [reply lib/utils/compose.js:537]: Yes

File: lib/utils/compose.js [hunk 3]
```diff
@@ -530,10 +530,13 @@ const getPreviousRepos = (sdk, docker, logger, appID) =>
 						return registry.imageName;
 					});
 				});
+			} else {
+				return Promise.resolve([]);
 			}
```
Comment (Page-) [inline lib/utils/compose.js:535]: There's no need for the promise.resolve here

File: lib/utils/compose.js [hunk 4]
```diff
@@ -530,10 +530,13 @@ const getPreviousRepos = (sdk, docker, logger, appID) =>
...
 		})
 		.catch(e => {
 			logger.logDebug(`Failed to access previously pushed image repo: ${e}`);
+			return Promise.resolve([]);
```
Comment (Page-) [inline lib/utils/compose.js:539]: There's no need for the promise.resolve here

**Comment by nazrhom** [comment]
@balena-ci retest

File: lib/utils/compose.js (hunk 1 above)
Comment (thgreasi) [reply lib/utils/compose.js:522]: I would also say that should be doing
```
.then(function(([release])) {
```

File: lib/utils/compose.js (hunk 3 above)
Comment (pdcastro) [reply lib/utils/compose.js:535]: Done

**Pull request merged**